src/pip/_internal/configuration.py

```diff
@@ -220,13 +219,7 @@ def unset_value(self, key):
 
             if modified_something:
                 # name removed from parser, section may now be empty
-                section_iter = iter(parser.items(section))
-                try:
-                    val = next(section_iter)
-                except StopIteration:
-                    val = None
-
-                if val is None:
+                if not parser.items(section):
```

Comment:
This can be problematic if `items()` returns an iterator, not a collection. The `next()`-based implementation is more correct. There might be an even better way to implement this though; I didn’t think too deeply.

Reply:
Makes sense,  I have reverted the change in https://github.com/pypa/pip/pull/8566/commits/4cc731c62b2e69254b96b7edb4844224f246e9e4